pkgs/tools/misc/direnv/default.nix

```diff
@@ -18,6 +18,11 @@ buildGoModule rec {
     stdenv.lib.optionalString (!stdenv.hostPlatform.isWindows)
     "${bash}/bin/bash";
 
+  # fix hardcoded GOFLAGS in makefile
+  postPatch = ''
+    substituteInPlace GNUmakefile --replace "export GOFLAGS=-mod=vendor" ""
+  '';
+
   # replace the build phase to use the GNUMakefile instead
```

Comment:
cc @zimbatm

Reply:
I created an issue upstream to better handle inheriting the env vars. Until then, the proposed solution is good.